internal/metrics/metrics.go

```diff
@@ -0,0 +1,88 @@
+// Copyright 2022 ChainSafe Systems (ON)
+// SPDX-License-Identifier: LGPL-3.0-only
+
+package metrics
+
+import (
+	"context"
+	"fmt"
+	"net/http"
+	"time"
+
+	"github.com/ChainSafe/gossamer/internal/httpserver"
+	"github.com/ChainSafe/gossamer/internal/log"
+	"github.com/prometheus/client_golang/prometheus/promhttp"
+)
+
+const defaultInterval = 10 * time.Second
+
+var logger log.LeveledLogger = log.NewFromGlobal(log.AddContext("pkg", "metrics"))
+
+// IntervalConfig for interval collection
+type IntervalConfig struct {
```

Comment:
Why not call it just `Config` though?

Reply:
I'd hope in most cases we wouldn't need any default metrics interval.  I'd rather the metrics be instrumented at the event.

Reply:
Ah I get it, this is only to configure the periodic metrics scrapping, this makes sense. Thanks!